src/components/checkbox/checkbox.test.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import { fireEvent, render, screen } from '@testing-library/react'
+import { Checkbox } from './checkbox'
+
+test('renders checkbox label correctly', () => {
+  let checked = false
+  render(
+    <Checkbox
+      label="Checkbox label"
+      checked={checked}
+      onChange={() => {
+        checked = !checked
+      }}
+    />,
+  )
+  expect(screen.getByText('Checkbox label')).toBeInTheDocument()
+})
+
+test('onChange event of Checkbox working correctly', () => {
+  let checked = false
+  render(
+    <Checkbox
+      label="Checkbox label"
+      checked={checked}
+      onChange={() => (checked = !checked)}
```

Comment:
`() => (...)` is meant to return something. As we are not returning anything, we should be using

```jsx
onChange={() => {
  checked = !checked
}
```

Reply:
I think we should be using something like

```jsx
onChange={() => {
  checked = event.target.checked
}}
```

as we are testing whether is toggly the checked state correctly or not

Reply:
Or better we should be doing something like this

```jsx
test('onChange event of Checkbox working correctly', () => {
  const onChange = jest.fn((event) => event.target.checked)
  render(<Checkbox label="Checkbox label" checked onChange={onChange} />)
  fireEvent.click(screen.getByText('Checkbox label'))
  expect(onChange).toBeCalled()
  expect(onChange.mock.results[0].value).toBe(false)
})
```